source/common/ssl.cc

```diff
@@ -0,0 +1,159 @@
+#include "common/ssl.h"
+
+namespace Nighthawk {
```

Comment:
This seems like it's getting really complicated. Do you think there is an upstream Envoy change we can make to simplify?

Reply:
Re: "This seems like it's getting really complicated. Do you think there is an upstream Envoy change we can make to simplify?"

These fakes are gone now. I'll ping you on Slack to discuss changes made here in the commits I just pushed